Sources/Controllers/2.0/OACustomPOIViewController.mm

```diff
@@ -131,93 +150,115 @@ - (void)updateShowButton:(BOOL)hasSelection
-    UIColor *colorShow = _countShowCategories != 0 ? UIColor.whiteColor : UIColorFromRGB(color_text_footer);
-    NSMutableAttributedString *attrShow = [[NSMutableAttributedString alloc] initWithString:textShow attributes:@{NSFontAttributeName: fontShow, NSForegroundColorAttributeName: colorShow}];
+        NSString *textShow = OALocalizedString(@"sett_show");
+        UIFont *fontShow = [UIFont systemFontOfSize:15 weight:UIFontWeightSemibold];
+        UIColor *colorShow = _countShowCategories != 0 ? UIColor.whiteColor : UIColorFromRGB(color_text_footer);
+        NSMutableAttributedString *attrShow = [[NSMutableAttributedString alloc] initWithString:textShow attributes:@{NSFontAttributeName: fontShow, NSForegroundColorAttributeName: colorShow}];
+
+        NSString *textCategories = [NSString stringWithFormat:@"\n%@: %li", OALocalizedString(@"categories"), _countShowCategories];
+        UIFont *fontCategories = [UIFont systemFontOfSize:13 weight:UIFontWeightRegular];
+        UIColor *colorCategories = _countShowCategories != 0 ? [[UIColor alloc] initWithWhite:1 alpha:0.5] : UIColorFromRGB(color_text_footer);
+        NSMutableAttributedString *attrCategories = [[NSMutableAttributedString alloc] initWithString:textCategories attributes:@{NSFontAttributeName: fontCategories, NSForegroundColorAttributeName: colorCategories}];
 
-    NSString *textCategories = [NSString stringWithFormat:@"\n%@: %li", OALocalizedString(@"categories"), _countShowCategories];
-    UIFont *fontCategories = [UIFont systemFontOfSize:13 weight:UIFontWeightRegular];
-    UIColor *colorCategories = _countShowCategories != 0 ? [[UIColor alloc] initWithWhite:1 alpha:0.5] : UIColorFromRGB(color_text_footer);
-    NSMutableAttributedString *attrCategories = [[NSMutableAttributedString alloc] initWithString:textCategories attributes:@{NSFontAttributeName: fontCategories, NSForegroundColorAttributeName: colorCategories}];
+        [attrShow appendAttributedString:attrCategories];
 
-    [attrShow appendAttributedString:attrCategories];
+        NSMutableParagraphStyle *style = [[NSMutableParagraphStyle alloc] init];
+        [style setLineSpacing:2.0];
+        [style setAlignment:NSTextAlignmentCenter];
+        [attrShow addAttribute:NSParagraphStyleAttributeName value:style range:NSMakeRange(0, attrShow.string.length)];
 
-    NSMutableParagraphStyle *style = [[NSMutableParagraphStyle alloc] init];
-    [style setLineSpacing:2.0];
-    [style setAlignment:NSTextAlignmentCenter];
-    [attrShow addAttribute:NSParagraphStyleAttributeName value:style range:NSMakeRange(0, attrShow.string.length)];
+        [self.showButton setAttributedTitle:attrShow forState:UIControlStateNormal];
+        [self updateShowButton:_countShowCategories != 0];
 
-    [self.showButton setAttributedTitle:attrShow forState:UIControlStateNormal];
-    [self updateShowButton:_countShowCategories != 0];
+        self.showBottomViewHeightPrimaryConstraint.active = !_keyboardVisible;
```

Comment:
Because depending on whether there is a search result on the screen or not, the text on the bottom button will change, respectively, its size and indentation from the bottom too